### PR TITLE
Add support for file extensions to django-storage backend.

### PR DESCRIPTION
**Pull request by Squirrel18**
## Description:

This PR adds support for file extensions to django-storage backend, in order to save and retrieve the correct file with the correct file extension.

This PR will no be merged, it is just for review.

## Reviewers:

- [ ] @felipemontoya 
- [ ] @diegomillan 
- [ ] @morenol

### CODE REVIEW

**Comment by Squirrel18** [comment]
@felipemontoya @diegomillan PR updated with .meta file support.

**Comment by felipemontoya** [comment]
With regards to the approach taken here I think this is good to go :+1: .
However a more formal code review on the line-per-line code changes is still required. Either @morenol or @diegomillan could do this

File: openassessment/fileupload/backends/django_storage.py [hunk 1]
```diff
@@ -44,6 +80,20 @@ def remove_file(self, key):
         Returns False if the file does not exist, and so was not removed.
         """
         path = self._get_file_path(key)
+        metadata_path = '{path}{ext}'.format(path=path, ext=self.METADATA_FILE_EXTENSION)
+
+        # Checks that .meta file exists for this key.
+        if default_storage.exists(metadata_path):
+            metadata_file = default_storage.open(metadata_path).read()
+            metadata_dict = json.loads(metadata_file)
```
Comment (diegomillan) [inline openassessment/fileupload/backends/django_storage.py]: We could create a function to obtain `metadata_dict` then we could use it here, and also above in `get_download_url()`. We would be avoiding code repetition.

**Review by diegomillan — CHANGES_REQUESTED** [review]

**Comment by diegomillan** [comment]
Minor suggestion @Squirrel18. Please, applied it and we will be ready to cherry-pick these changes.

**Review by diegomillan — APPROVED** [review]
LGTM now :+1:

File: openassessment/fileupload/backends/django_storage.py (hunk 1 above)
Comment (Squirrel18) [reply openassessment/fileupload/backends/django_storage.py]: Addresed @diegomillan

**Comment by Squirrel18** [comment]
PR closed, continues on Gitlab. @diegomillan @morenol @felipemontoya

**Pull request closed without merging**